src/cronjobs/legacy/price-migration.ts

```diff
@@ -0,0 +1,250 @@
+import { CronJob } from 'cron';
+import dayjs from 'dayjs';
+import { and, eq, lte, sql } from 'drizzle-orm';
+import gql from 'graphql-tag';
+import { uniqBy } from 'lodash';
+
+import { db } from 'database/client';
+import { tokenRepository } from 'database/repository/token-repository';
+import { flags, usdDailyPricesTable, usdHourlyPricesTable, usdPricesTable } from 'database/schema';
+import { networks } from 'loader/networks';
+import { LegacyChain } from 'loader/networks/legacy-chain';
+import { NetworkFeature } from 'loader/networks/types';
+import { CurrentPrice, prepareDataToStore, Price } from 'loader/usd-prices/usd-price-store';
+import { floorDate } from 'utils/date';
+import { getFlag, setFlag } from 'utils/flag';
+import { logger } from 'utils/logger';
+
+const childLogger = logger.child({ module: 'price-feed-task' });
+
+const BLOCKS = 25;
+
+export const priceFeedTask = async (ctx: CronJob) => {
+  ctx.stop();
+  childLogger.info('Price feed task start.');
+
+  try {
+    const items = networks.listChains();
+
+    for (const item of items) {
+      if (item.hasFeature(NetworkFeature.legacy)) {
+        await processLegacyChain(item.legacy);
+      }
+    }
+
+    childLogger.info('Price feed task ended.');
+    ctx.start();
+  } catch (error) {
+    childLogger.error({ error }, 'Price feed task failed.');
+    ctx.start();
+  }
+};
+
+const processLegacyChain = async (chain: LegacyChain) => {
+  const key = `price-feed-${chain.context.chainId}-2`;
+  const currentBlock = await chain.context.rpc.getBlockNumber();
+  const savedBlock = await getFlag(key).then((value) => (value ? Number(value) : 4580025)); // 4580025
```

Comment:
it's block since when we track prices (basically swap contract deployment block).
file is supposed to be removed later, but I put it as constants at the top of the file now